test/fixtures.py

```diff
@@ -63,14 +65,55 @@ def pylsp_w_workspace_folders(tmpdir):
     return (ls, workspace_folders)
 
 
+class FakeEditorMethodsMixin:
```

Comment:
Please move all the new supporting classes you added here to be before the fixture declarations.

Reply:
done